### PR TITLE
goRouter redirection feature for MacOS and ios version

### PR DESCRIPTION
**Pull request by molly-yan**

### CODE REVIEW

File: lib/providers/auth_provider.dart [hunk 1]
```diff
@@ -32,6 +34,21 @@ class AuthProvider extends ChangeNotifier {
...
+    print('stored token_read:$storedToken');
+    if (storedToken != null) {
+      _loginToken = Token.fromJson(json.decode(storedToken));
+      _loggedInStatus = Status.LoggedIn;
+    } else {
+      _loggedInStatus = Status.NotLoggedIn;
+    }
+    notifyListeners();
+  }
+
```
Comment (IanSaucy) [inline lib/providers/auth_provider.dart]: Similar to what I was talking to Jason about for the 519 project. We want to verify a token when it's retrieved from the storage. I think this library will help with that. Simply verify that it has an expiration date greater than the current date (be careful with timezones) 

https://pub.dev/packages/jwt_decoder

File: lib/utility/shared_pref.dart
Comment (IanSaucy) [inline lib/utility/shared_pref.dart:1]: Can you remind me why we need this and the `flutter_secure_storage`? Don't both of them persist the JWT?

**Review by IanSaucy — CHANGES_REQUESTED** [review]
Great work! Just a few clarifying questions

File: lib/utility/shared_pref.dart
Comment (molly-yan) [reply lib/utility/shared_pref.dart:1]: Sorry that I forgot to remove this file. I created it before learning about flutter_secure_storage.

File: lib/providers/auth_provider.dart [hunk 2]
```diff
@@ -39,11 +40,16 @@ class AuthProvider extends ChangeNotifier {
...
-      _loginToken = Token.fromJson(json.decode(storedToken));
-      _loggedInStatus = Status.LoggedIn;
-    } else {
+      DateTime expirationDate = JwtDecoder.getExpirationDate(storedToken);
+      //check if the token is expired or not
+      if (expirationDate.isAfter(DateTime.now())) { //still valid
+        _loginToken = Token.fromJson(json.decode(storedToken));
+        _loggedInStatus = Status.LoggedIn;
+      } else { //token expired
+        _loggedInStatus = Status.NotLoggedIn;
```
Comment (IanSaucy) [inline lib/providers/auth_provider.dart:53]: I would also clear the stored token here too.

**Review by IanSaucy — CHANGES_REQUESTED** [review]
one more change!

**Review by IanSaucy — APPROVED** [review]
lgtm

**Pull request merged**